src/rgw/driver/rados/rgw_reshard.cc

```diff
@@ -259,6 +293,13 @@ class BucketReshardShard {
     }
     entries.clear();
     stats.clear();
+
+    if (br != nullptr) {
+      ret = br->renew_lock_if_needed(dpp);
```

Comment:
> so i guess it's for BucketReshardManager::finish()

@cbodley  Yep, if I remember correctly, we found it will timeout to renew reshard_lock at the end of second stage for some tests. For example, there are 2048 source shards(maybe because with many rocksdb tombs), `finish()->flush()` is async for each shard, and `wait_all_aio()` may wait all 2048  source shards flushed. 
You are right, moving `renew_lock_if_needed()` to `finish()` is much better, and the codes can be more clean. I will move it to the loop of `shard.wait_all_aio()`.